fault_set.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2019 - 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goscaleio
+
+import (
+	// "errors"
+	"fmt"
+	"net/http"
+	// "time"
+
+	types "github.com/dell/goscaleio/types/v1"
+)
+
+// CreateFaultSet creates a fault set
+func (pd *ProtectionDomain) CreateFaultSet(fs *types.FaultSetParam) (string, error) {
+	path := fmt.Sprintf("/api/types/FaultSet/instances")
+	fs.ProtectionDomainID = pd.ProtectionDomain.ID
+	fsResp := types.FaultSetResp{}
+	err := pd.client.getJSONWithRetry(
+		http.MethodPost, path, fs, &fsResp)
+	if err != nil {
+		return "", err
+	}
+	return fsResp.ID, nil
+}
+
+// DeleteFaultSet will delete a fault set
+func (pd *ProtectionDomain) DeleteFaultSet(id string) error {
+	path := fmt.Sprintf("/api/instances/FaultSet::%v/action/removeFaultSet", id)
+	fsParam := &types.EmptyPayload{}
+	err := pd.client.getJSONWithRetry(
+		http.MethodPost, path, fsParam, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ModifyFaultSetName will modify the name of the fault set
+func (pd *ProtectionDomain) ModifyFaultSetName(id, name string) error {
+	fs := &types.FaultSetRename{}
+	fs.NewName = name
+	path := fmt.Sprintf("/api/instances/FaultSet::%v/action/setFaultSetName", id)
+
+	err := pd.client.getJSONWithRetry(
+		http.MethodPost, path, fs, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ModifyFaultSetName will modify the name of the fault set
+func (pd *ProtectionDomain) ModifyFaultSetPerFrofile(id, perfProfile string) error {
+	pp := &types.ChangeSdcPerfProfile{}
+	pp.PerfProfile = perfProfile
+	path := fmt.Sprintf("/api/instances/FaultSet::%v/action/setSdsPerformanceParameters", id)
+
+	err := pd.client.getJSONWithRetry(
+		http.MethodPost, path, pp, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ModifyFaultSetName will modify the name of the fault set
+func (pd *ProtectionDomain) ReadFaultSet(id string) (*types.FaultSet, error) {
```

Comment:
Please change the comment as per the method name

Reply:
Done